lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -6447,29 +6424,31 @@ SILValue ADContext::promoteToDifferentiableFunction(
             thunk->isBare(), thunk->isTransparent(), thunk->isSerialized(),
             thunk->isDynamicallyReplaceable(), ProfileCounter(),
             thunk->isThunk());
+        // If new thunk does not exist: clone the old thunk body, wrap the
+        // returned function value with an `autodiff_function` instructions,
+        // and process the `autodiff_function` instruction.
         if (newThunk->empty()) {
           newThunk->setOwnershipEliminated();
           SILFunctionCloner cloner(newThunk);
           cloner.cloneFunction(thunk);
-        }
 
-        auto *retInst =
-            cast<ReturnInst>(newThunk->findReturnBB()->getTerminator());
-        AutoDiffFunctionInst *adfi;
-        {
-          SILBuilder builder(retInst);
-          adfi = createAutoDiffFunction(builder, loc, parameterIndices,
-                                        differentiationOrder,
-                                        retInst->getOperand());
+          auto *retInst =
+              cast<ReturnInst>(newThunk->findReturnBB()->getTerminator());
```

Comment:
This will segfault on functions that do not have a return block. Please add tests for functions that use `fatalError` to ensure things work.

Reply:
This logic triggers only for curry thunks, which are guaranteed to have a return block/instruction.
I can't think of a way to add tests.